includes/class-sensei-lesson.php

```diff
@@ -347,9 +347,13 @@ private function output_prerequisite_meta_box_content( int $lesson_id, int $cour
 				$html .= '<option value="' . esc_attr( absint( $post_item->ID ) ) . '"' . selected( $post_item->ID, $selected_prerequisite, false ) . '>' . esc_html( $post_item->post_title ) . '</option>' . "\n";
 			}
 			$html .= '</select>' . "\n";
-		} else {
+		} elseif ( ! $course_id ) {
 			$html .= '<input type="hidden" name="' . esc_attr( $input_name ) . '" value="">';
-			$html .= '<p>' . esc_html__( 'No lessons exist yet. Please add some first.', 'sensei-lms' ) . '</p>';
+			$html .= '<p>' . esc_html__( 'Please select a course first.', 'sensei-lms' ) . '</p>';
+		} else {
+			$edit_course_url = esc_url( sprintf( 'post.php?post=%d&action=edit', $course_id ) );
+			$html           .= '<input type="hidden" name="' . esc_attr( $input_name ) . '" value="">';
+			$html           .= '<p>' . esc_html__( 'No lessons exist yet. Please add some to ', 'sensei-lms' ) . '<a href="' . $edit_course_url . '">' . esc_html__( 'the course', 'sensei-lms' ) . '</a>.</p>';
```

Comment:
For links, it's preferable to embed them directly in the string. Currently the sentence is split in two, which will make it difficult for translators because they'll be missing the context from the rest of the sentence.

Then, because we're embedding HTML (i.e. the anchor tag) in the string, we can't use `esc_html__` anymore because that will escape it. We'll need to use `wp_kses_post` instead.

So it would end up looking something like this:
```
$html .= '<p>' . wp_kses_post( sprintf( __( 'No lessons exist yet. Please add some to <a href="%s">the course</a>.', 'sensei-lms' ), esc_url( $edit_course_url ) ) ) . '</p>';
```

Note that I also moved `esc_url` from the variable to the string to follow WordPress' best practice of [escaping late](https://wpvip.com/2014/06/20/escaping-output/).

The PHP linter will probably also complain about adding a [translator comment](https://codex.wordpress.org/I18n_for_WordPress_Developers#Descriptions) to explain what `%s` means. You can see an example of how to do that [here](https://github.com/Automattic/sensei/blob/83634001c080a8b8b5f090abc20403b9967cfeb7/includes/class-sensei-posttypes.php#L810).

Reply:
Ohh good to know about ```wp_kses_post```, yeah I was looking for something like that. Thanks